backend/inspirehep/search/api.py

```diff
@@ -177,7 +176,15 @@ def match_reference(self, query_string):
         except KeyError:
             pass
 
-        return match_reference_control_number(reference)
+        reference_match_control_number = match_reference_control_number(reference)
```

Comment:
you could also get the full result here, as the matching is already doing a search (actually several searches), but it adds more complexity, so this is probably good enough.

Reply:
Indeed, but I'd have to modify also the matcher api which we suppose to use them for `next` as well. By default it returns the reference with the record `$ref`